core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -243,7 +243,7 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   private[spark] def eventLogDir: Option[URI] = _eventLogDir
   private[spark] def eventLogCodec: Option[String] = _eventLogCodec
 
-  def isLocal: Boolean = (master == "local" || master.startsWith("local["))
+  def isLocal: Boolean = Utils.isLocal(_conf)
```

Comment:
Keep using def, because val will cause test fails (val will be evaluated when declared while def will be evaluated when invoked ) Although I can use config instead of _conf, but considering all the variables around isLocal is using _conf, I don't want to make it inconsistent and involve any potential issue.